apps/system/lockscreen/js/source/minute_clock_source.js

```diff
@@ -0,0 +1,55 @@
+/* global SourceEvent */
+'use strict';
+
+/**
+ * A source fire events every clock minutes.
+ **/
+(function(exports) {
+  var MinuteClockSource = function(configs) {
+    this.configs = {
+      type: configs.type,
+      interval: 60000       // one minute.
+    };
+    this._id = null;
+    this._forwardTo = null;
+    // Some API you just can't bind it with the object,
+    // but a function.
+    this.onchange = this.onchange.bind(this);
+  };
+
+  MinuteClockSource.prototype.start = function(forwardTo) {
+    this._forwardTo = forwardTo;
+    var seconds = (new Date()).getSeconds();
+    // If it's at the second 0th of the minute, immediate start to tick.
+    var leftMilliseconds = (0 === seconds) ? 0 : (60 - seconds) * 1000;
+    window.setTimeout(() => {
+      this.onchange();    // When it's up to the 0 sec of the new minute.
+      this._id = window.setInterval(() => {
+        this.onchange();
+      }, this.configs.interval);
```

Comment:
After our discussion, I recalled that there is a bug to change the `setTimeout` style to `setInterval` because of the performance issue. So I want to reduce the risk, and may change it after we land this patch.